### DISCUSSION TITLE
Should breaking change in PR #1200 raise error instead of emitting warning?

### DISCUSSION THREAD

**Discussion by bradday4** [PyO3/maturin#1460] [Q&A] [answered]
The breaking change in https://github.com/PyO3/maturin/pull/1200 for python package metadata in `cargo.toml` is emitted as a warning instead of an error. i.e the build system continues despite the fact that python dependencies will not be incorporated into the project. I know of 1 project where this has caused a bad build artifact from their CI/CD pipeline [bed-reader](https://pypi.org/project/bed-reader/) . Wouldn't it be better to have Maturin raise an exception instead of continuing the build process if cargo.toml still defines these [keys](https://github.com/PyO3/maturin/blob/46f4b1a0c1e4583de998467e8579ad2b69a0d685/src/cargo_toml.rs#L63-L75)?

**Comment by messense** [comment, marked as answer]
Implemented in https://github.com/PyO3/maturin/pull/1471.

`bed-reader` should add a version constraint in https://github.com/fastlmm/bed-reader/blob/46ece2e6efd1fb8f5a9335fd8b3ef7a3d711b9fa/.github/workflows/ci.yml#L60, same as https://github.com/fastlmm/bed-reader/blob/46ece2e6efd1fb8f5a9335fd8b3ef7a3d711b9fa/pyproject.toml#L20
